### PR TITLE
Making python3 compatible

### PR DESCRIPTION
**Pull request by samsteyer**
Small changes to make `sscpy` compatible with Python 3.

### CODE REVIEW

File: ssc/api.py [hunk 1]
```diff
@@ -106,13 +107,19 @@ def keys(self):
             key = _LIB.ssc_data_next(self._data)
         return keys_
 
+    def __iter__(self):
+        pass
```
Comment (jerluc) [inline ssc/api.py]: This should be implemented

File: ssc/api.py [hunk 2]
```diff
@@ -106,13 +107,19 @@ def keys(self):
...
+
+    def __len__(self):
+        pass
```
Comment (jerluc) [inline ssc/api.py]: This should also be implemented

File: ssc/api.py [hunk 3]
```diff
@@ -43,8 +43,9 @@ def __init__(self):
 
     def _get_data_type(self, name):
         for key in self.keys():
+            key = key.decode('ascii')
```
Comment (jerluc) [inline ssc/api.py]: Why does this require ASCII? I think this should be taking `utf-8` if at all possible.

**Review by jerluc — CHANGES_REQUESTED** [review]
Needs to be reworked for UTF-8/unicode support and also to fully implement the dict interface for the `Data` class.

File: ssc/api.py (hunk 3 above)
Comment (samsteyer) [reply ssc/api.py]: Ah, that makes more sense

**Review by jerluc — APPROVED** [review]
Nice! Looks a lot clearer now that we are explicitly casting these to unicode `bytes` values. Assuming this all passes your own manual testing, this should be good to go.

We need to implement some unit tests eventually :(

**Pull request merged**